frontend/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,95 @@
+// React Framework imports
+import * as React from "react";
+import { Link } from "react-router-dom"
+import clsx from 'clsx';
+
+// Material-UI Functionality imports
+import Divider from '@material-ui/core/Divider';
+import Drawer from '@material-ui/core/Drawer';
+import IconButton from '@material-ui/core/IconButton';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { createStyles, makeStyles, 
+         useTheme, Theme } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+
+// Material-UI Icon imports
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import ExploreIcon from '@material-ui/icons/Explore';
+import FaceIcon from '@material-ui/icons/Face';
+import LibraryBooksIcon from '@material-ui/icons/LibraryBooks';
+
+const drawerWidth = 240;
```

Comment:
If you do end up going this route, make this an exported const from another file (maybe something like `layout_constants.ts`) so that it doesn't appear both here and the app bar. In general we want to make sure every shared value lives in only one place so it can be easily changed later (eg. if we realize 240 isn't wide enough to fit the Bulgarian navigation text and need to make it wider for everybody, we don't have to search through our codebase to find every instance of `240` and just hope they're all related to the nav bar width)

Reply:
Very valid point!